tfx/dsl/component/experimental/function_parser.py

```diff
@@ -21,14 +21,19 @@
 
 import enum
 import inspect
+import sys
 import types
 from typing import Any, Dict, Optional, Tuple, Type, Union
-
 from tfx.dsl.component.experimental import annotations
 from tfx.dsl.component.experimental import json_compat
 from tfx.types import artifact
 from tfx.types import standard_artifacts
 
+if sys.version_info >= (3, 8):
+  from typing import _TypedDictMeta
+else:
+  from typing_extensions import _TypedDictMeta
```

Comment:
Hello, this imports (importing private classes starting from underscores (_)) caused some internal test failures.
Could you change them into public classes imports?